src/Grpc.Net.Client/ChannelBuilder.cs

```diff
@@ -0,0 +1,152 @@
+﻿#region Copyright notice and license
+
+// Copyright 2019 The gRPC Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#endregion
+
+using System;
+using System.Diagnostics;
+using System.Net.Http;
+using Grpc.Core;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+
+#pragma warning disable CS1591 // Missing XML comment for publicly visible type or member
+
+namespace Grpc.Net.Client
+{
+    public sealed class ChannelBuilder
+    {
+        private GrpcChannelOptions _options;
+        private ILoggerFactory? _loggerFactory;
+
+        private ChannelBuilder() : this(new GrpcChannelOptions())
+        {
+        }
+
+        private ChannelBuilder(GrpcChannelOptions options)
+        {
+            _options = options;
+        }
+
+        public static ChannelBuilder ForUrl(Uri uri)
```

Comment:
I wonder if we want something we do for WebHost along the lines of WebHost.CreateDefaultBuilder(...). I can see GrpcChannel.CreateDefaultBuilder(...) or GrpcChannel.CreateBuilder(...).